commands/ls.go

```diff
@@ -185,3 +191,22 @@ func matchesState(host *libmachine.Host, states []string) bool {
 	}
 	return false
 }
+
+func matchesName(host *libmachine.Host, names []string) bool {
+	if len(names) == 0 {
+		return true
+	}
+	for _, n := range names {
+		r, err := regexp.Compile(n)
+		if err != nil {
+			log.Warn(err)
```

Comment:
If the regex doesn't compile, we should probably just bail (`log.Fatal`), right?  Otherwise it might lead to unexpected behavior.  @ehazlett @hairyhenderson